consensus/istanbul/backend/backend.go

```diff
@@ -156,14 +161,23 @@ func (sb *backend) Gossip(valSet istanbul.ValidatorSet, payload []byte) error {
 
 			m.Add(hash, true)
 			sb.recentMessages.Add(addr, m)
-			go p.SendConsensus(istanbulMsg, payload)
+
+			var outboundCode uint64 = istanbulMsg
+			if _, ok := qbftMessage.MessageCodes()[code]; ok {
+				outboundCode = code
+			}
+			if sb.IsQBFTConsensus() {
+				go p.SendQbftConsensus(outboundCode, payload)
+			} else {
+				go p.SendConsensus(outboundCode, payload)
+			}
```

Comment:
I don't think it is clear that for non QBFT, the `code` is still `istanbulMsg`.
Here it will ensure it is the case:
```suggestion

			if sb.IsQBFTConsensus() {
				var outboundCode uint64 = istanbulMsg
				if _, ok := qbftMessage.MessageCodes()[code]; ok {
					outboundCode = code
				}
				go p.SendQbftConsensus(outboundCode, payload)
			} else {
				go p.SendConsensus(istanbulMsg, payload)
			}
```

Reply:
Done